report/validation/surface_model/markelj/markelj_DdamagedW.md

```diff
@@ -0,0 +1,462 @@
+)
+
+TXT = F.TXTExport(field="retention", filename="./FESTIM_sim.txt", times=export_times)
+
+W_model.exports = [derived_quantities] + [TXT]
+
+W_model.dt = F.Stepsize(
+    initial_value=1e-5,
+    stepsize_change_ratio=1.25,
+    max_stepsize=500,
+    dt_min=1e-6,
+    milestones=export_times,
+)
+
+W_model.settings = F.Settings(
+    absolute_tolerance=1e7,
+    relative_tolerance=1e-13,
+    final_time=final_time,
+    traps_element_type="DG",
+)
+
+W_model.initialise()
+W_model.run()
+```
+
+## Comparison with experimental data and MHIMS: D content
+
++++
+
+The results produced by FESTIM are in good agreement with the experimental data and correlate perfectly with MHIMS.
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+import plotly.graph_objects as go
+import plotly.express as px
+from scipy.interpolate import interp1d
+
+
+def RMSPE(x_sim, x_exp):
+    error = np.sqrt(np.mean((x_sim - x_exp) ** 2)) / np.mean(x_exp)
+    return error
+
+
+retention = (
+    np.array(derived_quantities[0].data)
+    + np.array(derived_quantities[1].data)
+    + np.array(derived_quantities[2].data)
+)
+t = np.array(derived_quantities.t)
+
+exp_ret = np.loadtxt("./reference_data/exp_ret.csv", delimiter=",", skiprows=1)
+
+interp_ret = interp1d(t / 3600, retention, fill_value="extrapolate")
+
+error = RMSPE(
+    interp_ret(
+        exp_ret[:, 0],
+    ),
+    exp_ret[:, 1],
+)
+
+print(f"RMSPE between FESTIM and experimental data is {error*100:.2f}%")
+
+MHIMS_ret = np.loadtxt("./reference_data/MHIMS_ret.csv", delimiter=",", skiprows=1)
+
+fig = go.Figure()
+
+fig.add_trace(
+    go.Scatter(
+        x=t / 3600,
+        y=retention / 1e20,
+        mode="lines",
+        line=dict(width=3, color=px.colors.qualitative.Plotly[0]),
+        name="FESTIM",
+    )
+)
+
+fig.add_trace(
+    go.Scatter(
+        x=MHIMS_ret[::100, 0],
+        y=MHIMS_ret[::100, 1] / 1e20,
+        mode="markers",
+        marker_symbol="square",
+        marker=dict(size=9, color=px.colors.qualitative.Plotly[2]),
+        name="MHIMS",
+    )
+)
+
+fig.add_trace(
+    go.Scatter(
+        x=exp_ret[:, 0],
+        y=exp_ret[:, 1] / 1e20,
+        mode="markers",
+        marker=dict(size=10, color=px.colors.qualitative.Plotly[1]),
+        name="Exp.",
+    )
+)
+
+fig.update_yaxes(
+    title_text="D inventory, 10<sup>20</sup> m<sup>-2</sup>",
+    range=[0, 5],
+    tick0=0,
+    dtick=1,
+)
+fig.update_xaxes(title_text="Time, h", range=[0, 100], tick0=0, dtick=10)
+fig.update_layout(template="simple_white", height=600)
+
+# The writing-reading block below is needed to avoid the issue with compatibility
+# of Plotly plots and dollarmath syntax extension in Jupyter Book
+# For mode details, see https://github.com/jupyter-book/jupyter-book/issues/1528
+
+fig.write_html("./markelj_comparison_ret.html")
+from IPython.display import HTML, display
+
+display(HTML("./markelj_comparison_ret.html"))
+```
+
+## Comparison with experimental data and MHIMS: D depth distribution
```

Comment:
This figure is very cluttered and since many lines overlap I think that we should either:
- add a slider with the time and only show the current timestep
- make subplots